src/server/robot/grid-indices.ts

```diff
@@ -0,0 +1,30 @@
+import { Pair } from "../utils/pair";
+
+/**
+ * Defines a position on the 12x12 grid robots are allowed to be in.
+ */
+export class GridIndices extends Pair<GridIndices> {
+    constructor(
```

Comment:
Can we make this constructor private?

Reply:
There might be use cases where direct construction is good for tests? IDK, private constructor just seems kinda smelly to me. Maybe I just have trauma from singletons

Reply:
Well I would imagine you don't ever want to construct a gridindices bypassing the index checks in any normal use